### DISCUSSION TITLE
Share your bigtree projects!

### DISCUSSION THREAD

**Discussion by kayjan** [kayjan/bigtree#52] [Show and tell]
Thank you for using `bigtree`. Feel free to share your use case or project here!

**Comment by mvsoliveira** [comment]
Hi Kay,

Great idea. I am using bigtree for Wishlist, which is a project I am working on at CERN aiming to systematically ensure consistency between software and firmware register descriptions by generating, for example, address decoding and register description using computer-generated code. 

The project is hosted temporarily here:
https://gitlab.cern.ch/msilvaol/wishlist
I am planning to write documentation in the next 1-2 months.

**Comment by kj3moraes** [comment]
Hello,

I'm building a markup language parser in Python. It parses languages like Markdown, reStructuredText, AsciiDoc, YAML, etc. into forest data structures and manipulated from there onwards. They can also be converted back to their text forms. 

It is still in development and a lot of the features mentioned above are missing - [delorean](https://github.com/kj3moraes/delorean

**Comment by mr-nice** [comment]
Hey @kayjan,

I use bigtree in my [solutions-fuse](https://gitlab.com/mrnice/solution-fuse) project. The project is highly experimental and mounts a Microsoft Solution file (.sln) into a user file system, at least on Linux :D. The project reads the .sln file, parses all the vcxproj.filters files and represents them as files and directories in user space. The structure inside of the mounted filesystem is the same as the structure you would get inside Visual Studio. The files are accessible as symlinks. It is planned to use cp, move and so on to alter the project files. To cache the filesystem representation I parse the files and put them into a bigtree. The attributes allows me to keep all the required information in cache.

Thanks for this awesome project :)

**Comment by pgarrison** [comment]
Hi Kay,

We're using bigtree to model mother-daughter relationships between cells as they go through mitosis for an upcoming paper from the Allen Institute for Cell Science. (Repo is private for the moment.) If you would like us to cite bigtree in the journal article, could you share a preferred citation to use (could be one of your Medium articles)?

**Reply by kayjan** [reply]
Hello, happy to hear you found bigtree useful! There is no paper currently, however, feel free to cite the Medium article! 😄
